tests/dort_test.py

```diff
@@ -0,0 +1,202 @@
+        first_model_proto.graph.node.extend(next_model.graph.node)
+        first_model_proto.functions.extend(next_model.functions)
+        first_model_proto.graph.initializer.extend(next_model.graph.initializer)
+        first_model_proto.opset_import.extend(next_model.opset_import)
+
+        return first_model_proto
+
+    options = torch.onnx._OrtBackendOptions(  # pylint: disable=protected-access
+        export_options=export_options,
+        ort_session_options=ort_session_options,
+        pre_ort_model_transforms=[inline_function],
+    )
+
+    return torch.onnx._OrtBackend(options=options)  # pylint: disable=protected-access
+
+
+class FuncModule(torch.nn.Module):
+    def __init__(self, f, params=None):
+        if params is None:
+            params = ()
+        super().__init__()
+        self.f = f
+        self.ppp = torch.nn.Parameter(torch.Tensor([1]))
+        self.params = torch.nn.ParameterList(list(params))
+
+    def forward(self, *args):
+        f_args = list(itertools.chain(args, self.params))
+        f_args[0] = f_args[0] + self.ppp
+        res = self.f(*f_args)
+        return res
+
+
+class FuncModuleModule(torch.nn.Module):
+    def __init__(self, f):
+        super().__init__()
+        self.f = f
+        self.mod = f
+        self.ppp = torch.nn.Parameter(torch.Tensor([1]))
+
+    def forward(self, *args):
+        x = args[0] + self.ppp
+        res = self.mod(x, *args[1:])
+        return res
+
+
+class TestSimpleDort(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        torch._dynamo.reset()  # pylint: disable=protected-access
+        # hides the warnings
+        # torch.onnx.dynamo_export only implements opset version 18 for now.
+        # torch.library.impl_abstract was renamed to torch.library.register_fake.
+        warnings.simplefilter("ignore", (UserWarning, DeprecationWarning))
+
+    def assertONNX(
+        self,
+        f,
+        args,
+        onnx_export: str,
+        params=None,
+        fullgraph: bool = True,
+        atol=1e-6,
+        rtol=1e-6,
+        dynamic_axes=None,
+    ):
+        if sys.platform == "win32":
+            raise unittest.SkipTest("Windows not supported yet.")
+        assert isinstance(onnx_export, str), f"Export onnx is wrong for f={f}"
+        if isinstance(args, torch.Tensor):
+            args = [args]
+        if params is None:
+            params = ()
+        if isinstance(f, torch.nn.Module):
+            model = FuncModuleModule(f)
+        else:
+            model = FuncModule(f, params)
+        model.eval()
+
+        # forward/backward
+        local_aot_ort = make_aot_ort(dynamic=dynamic_axes is not None)
+
+        compiled_model = torch.compile(
+            copy.deepcopy(model),
+            backend=local_aot_ort,
+            dynamic=dynamic_axes is not None,
+            fullgraph=fullgraph,
+        )
+
+        baseline_result = model(*args)
+        result = compiled_model(*args)
```

Comment:
@xadupre I tried to put everything on CUDA and run the code, but even though compiled_model is on CUDA, this result will not be on CUDA. Do you know why?